src/gpu/intel/jit/ir/ir_builder.hpp

```diff
@@ -29,29 +28,37 @@ namespace gpu {
 namespace intel {
 namespace jit {
 
-class ir_builder_t {
+class idx_dispatcher_t {
 public:
-    const stmt_t &stmt() const { return stmt_; }
+    using vars_t = std::array<expr_t, 3>;
 
-    static const std::array<expr_t, 3> &local_ids() {
-        static thread_local std::array<expr_t, 3> vars
-                = {var_t::make(type_t::u16(), "local_id0"),
-                        var_t::make(type_t::u16(), "local_id1"),
-                        var_t::make(type_t::u16(), "local_id2")};
```

Comment:
Is there a reason these variables cannot just static constants? If they are immutable, `thread_local` should be immaterial.
```
static const std::array<const expr_t, 3> &local_ids() {
    static const std::array<const expr_t, 3> vars
    ...
}
```